File: experiments/runs/2026-09-20_08-53-35_claim-office-java-example-mapping_exact-ptdd-v1.1-refactor-subagent-pi_gpt-5-6-sol-codex-5/src/main/java/InsuranceValue.java
final class InsuranceValue {
    private InsuranceValue() { }

    static int forItemType(String itemType) {
        return switch (itemType) {
            case "sword" -> 1000;
            case "amulet" -> 600;
            case "staff" -> 800;
            case "potion" -> 400;
            case "rune", "moonstone" -> 250;
            default -> 0;
        };
    }
}

File: experiments/runs/2026-09-24_04-57-22_claim-office-example-mapping_exact-ptdd-v1.1-refactor-subagent-pi_gpt-6-sol-codex/src/insurance-value.ts
const AMULET_INSURANCE_VALUE = 600;
const SWORD_INSURANCE_VALUE = 1000;
const STAFF_INSURANCE_VALUE = 800;
const POTION_INSURANCE_VALUE = 400;
const COMPONENT_INSURANCE_VALUE = 250;

type Item = { type: string };

function isComponent(item: Item): boolean {
  return item.type === 'rune' || item.type === 'moonstone';
}

export function insuranceValueFor(item: Item): number {
  if (item.type === 'sword') return SWORD_INSURANCE_VALUE;
  if (item.type === 'staff') return STAFF_INSURANCE_VALUE;
  if (item.type === 'potion') return POTION_INSURANCE_VALUE;
  if (isComponent(item)) return COMPONENT_INSURANCE_VALUE;
  return AMULET_INSURANCE_VALUE;
}
